### PR TITLE
Fix event id calculation

### PR DESCRIPTION
**Pull request by richvdh**
Fix the event id calculated for events in v3 etc rooms: we have to calculate
the hash ~~*before*~~ *after* signing.

Turns out this was working by good fortune before (we never actually checked
the event ids).

### CODE REVIEW

File: lib/SyTest/Federation/Datastore.pm [hunk 1]
```diff
@@ -190,27 +190,30 @@ sub create_event
...
-         # calculate the event id from the hash
-         $event_id = id_for_event( $event, $room_version );
       }
+      $self->sign_event( $event );
+   } else {
+      die "event with explicit event_id in room v$room_version"
+         if defined $event_id;
+
+      $self->sign_event( $event );
+      $event_id = id_for_event( $event, $room_version );
```
Comment (erikjohnston) [inline lib/SyTest/Federation/Datastore.pm]: Isn't this now calculating the hash after signing?

Comment (richvdh) [reply lib/SyTest/Federation/Datastore.pm]: that's what I (wish I) said

Comment (erikjohnston) [reply lib/SyTest/Federation/Datastore.pm]: Why does it matter if we hash before or after we sign? The signatures should be stripped before we calculate the hash?

Comment (richvdh) [reply lib/SyTest/Federation/Datastore.pm]: yes, but the `hashes` aren't.

Comment (erikjohnston) [reply lib/SyTest/Federation/Datastore.pm]: Oh, and `sign_event` also adds the hash. Right, got you. So this basically says we need to calculate the event ID hash *after* we've added the content hashes.

**Pull request merged**